interfaces/cython/cantera/reaction.pyx

```diff
@@ -795,6 +795,44 @@ cdef class Reaction:
             if products:
                 self.products = products
 
+    def __init__(self, reactants=None, products=None, *, equation=None, rate=None,
+                 Kinetics kinetics=None, init=True, legacy=False, **kwargs):
+
+        if legacy:
+            return
+
+        if reactants and products and not equation:
+            equation = self.equation
```

Comment:
This leaves the object in an interesting state. Should there be any tests? I.e. what does calling the `rate` property getter do?

Reply:
Per your comment above, requiring rate as a parameter would resolve this edge case

Reply:
Right now, this gives you a `nan`-initialized `ArrheniusRate` object. I think that ultimately stems from the fact that not specifying a reaction `type` implies an elementary/Arrhenius rate.